mount/mount.go

```diff
@@ -189,6 +189,42 @@ func (d *Datastore) Put(key ds.Key, value []byte) error {
 	return cds.Put(k, value)
 }
 
+// Sync implements Datastore.Sync
+func (d *Datastore) Sync(prefix ds.Key) error {
+	// Sync all mount points below the prefix
+	// Sync the mount point right at (or above) the prefix
+	// Return ErrNoMount if no mount points are synced
+	atLeastOneSync := false
+	for _, m := range d.mounts {
```

Comment:
Can we use `lookupAll`?

Reply:
np, it might do a few more comparisons but it'll look much cleaner

Reply:
lmk how it looks now